### ISSUE TITLE
Support to SoapHeader

### ISSUE THREAD

**Issue by giraz82** [Ricorocks-Digital-Agency/Soap#21]
Hi, and thank you for this great library.
Unfortunately, I would need suppport to `SoapHeader`, to set an authentication header, but, I'm not able to manage it with this facade.
Is there any way to get it?
The final result on XML should be:
```xml
<soap:Header>
  <AuthHeader xmlns=".../">
    <Username>user</Username>
    <Password>password</Password>
  </AuthHeader>
</soap:Header>
```
Thank you in advance

**Comment by nedwors** [comment]
Hi!

Thanks for getting in touch and raising an issue, it's much appreciated. We will look to have this implemented in the coming week, will drop a comment when we have done so.

**Comment by giraz82** [comment]
I started to implement something working. I can open a PR so you will have a starting point for this feature, what do you think?

**Comment by lukeraymonddowning** [comment]
@giraz82 please do!

**Comment by giraz82** [comment]
@lukeraymonddowning here we are

**Comment by ArthurPatriot** [comment]
Need this feature ASAP!

**Comment by nedwors** [comment]
We have opened #23 for this feature.

**Comment by giraz82** [comment]
Really nice job, it works like a charm within my code.

**Comment by lukeraymonddowning** [comment]
Thos has now been merged and released in 1.3.0
